tor_core/initialize.py

```diff
@@ -211,17 +213,67 @@ def initialize(config):
     logging.debug('Gifs loaded.')
 
 
-def build_bot(name, log_name='transcribersofreddit.log', require_redis=True):
+def get_heartbeat_port(config):
+    """
+    Attempts to pull an existing port number from the filesystem, and if it
+    doesn't find one then it generates the port number and saves it to a key
+    file.
+
+    :param config: the global config object
+    :return: int; the port number to use.
+    """
+    try:
+        # have we already reserved a port for this process?
+        port = int(open('heartbeat.port', 'r').readline().strip())
```

Comment:
Not quite sure how Python handles this form of `open()`, but it looks like it might leave an open file descriptor which is only closed when the program does some garbage collection or exits. ([see also](https://stackoverflow.com/questions/9282967/how-to-open-a-file-using-the-open-with-statement))

It might be safer to do something like this:

```python
with open('heartbeat.port', 'r') as port_file:
    port = int(port_file.readline().strip())
```

Otherwise, if you don't use the `with` keyword, you would have to call `close()` on the variable returned from `open()` in order to close the file descriptor again.